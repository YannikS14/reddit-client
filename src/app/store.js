import { configureStore } from '@reduxjs/toolkit';
import postsReducer from '../features/posts/postsSlice';
import subredditReducer from '../features/subreddits/subredditsSlice';
import searchFieldReducer from '../features/searchField/searchFieldSlice';

export const store = configureStore({
  reducer: {
    posts: postsReducer,
    subreddits: subredditReducer,
    searchField: searchFieldReducer,
  },
});
