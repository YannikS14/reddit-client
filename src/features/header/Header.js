import React from 'react';
import SearchField from '../searchField/SearchField';

export default function Header() {
  const toggleDarkMode = (e) => {
    const htmlTag = document.getElementsByTagName('html');
    const lightModeBtn = document.getElementById('light-mode');
    const darkModeBtn = document.getElementById('dark-mode');

    if (e.target.id === 'light-mode') {
      htmlTag[0].classList.remove('dark');
      lightModeBtn.classList.add('text-primary');
      darkModeBtn.classList.remove('text-primary');
    } else {
      htmlTag[0].classList.add('dark');
      darkModeBtn.classList.add('text-primary');
      lightModeBtn.classList.remove('text-primary');
    }
  };

  return (
    <header className="fixed z-20 min-w-full shadow-lg bg-white dark:bg-gray-700 dark:text-gray-50">
      <div className="container mx-auto px-4 h-20 flex justify-between items-center">
        <div id="logo" className="flex items-center">
          <img
            className="w-8 fill-current text-green-600"
            src="reddit-logo.svg"
            alt="reddit logo"
          />
          <h1 className="pl-2 sm:text-xl font-semibold">
            <span className="font-bold">Reddit</span> Client
          </h1>
        </div>
        <SearchField />
        <div
          id="dark-mode-toggle"
          className="hidden sm:flex border-solid border-2 border-gray-300 dark:border-gray-500 rounded-md divide-x-2 divide-gray-300 dark:divide-gray-500 divide-solid"
        >
          <button
            id="light-mode"
            className="px-4 py-2 text-primary"
            onClick={(e) => toggleDarkMode(e)}
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-6 w-6 pointer-events-none"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
              />
            </svg>
          </button>
          <button
            id="dark-mode"
            className="px-4 py-2"
            onClick={(e) => toggleDarkMode(e)}
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              className="h-6 w-6 pointer-events-none"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
              />
            </svg>
          </button>
        </div>
      </div>
    </header>
  );
}
