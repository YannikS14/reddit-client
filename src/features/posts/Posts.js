import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { selectPosts, fetchPosts } from './postsSlice';
import Post from '../post/Post';

export default function Posts() {
  const { posts, isLoading, hasErrors } = useSelector(selectPosts);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchPosts());
  }, [dispatch]);

  return (
    <section className="sm:w-2/3 mx-4 flex-auto">
      {isLoading && (
        <h2 className="mb-8 p-6 bg-white rounded-md shadow-lg text-xl font-bold">
          Loading posts...
        </h2>
      )}
      {hasErrors && (
        <h2 className="mb-8 p-6 bg-white rounded-md shadow-lg text-xl text-red-500 font-bold">
          Cannot display posts...
        </h2>
      )}
      {posts &&
        posts.map((post) => {
          return <Post post={post} key={post.data.id} />;
        })}
    </section>
  );
}
