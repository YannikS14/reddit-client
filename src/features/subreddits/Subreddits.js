import React from 'react';
import { useSelector } from 'react-redux';
import { selectSubreddits } from './subredditsSlice';
import Subreddit from '../subreddit/Subreddit';

export default function Subreddits() {
  const { subreddits, activeSubreddit } =
    useSelector(selectSubreddits);

  return (
    <section className="sm:mx-4 sm:w-1/3 sm:ml-0 flex-auto">
      <div className="p-6 bg-white sm:rounded-md shadow-lg">
        <h2 className="text-xl font-bold">Subreddits</h2>
        <div className="mt-4">
          {subreddits.map((subreddit) => {
            return (
              <Subreddit
                subreddit={subreddit}
                key={subreddit.title}
                active={
                  activeSubreddit === subreddit.title ? true : false
                }
              />
            );
          })}
        </div>
      </div>
    </section>
  );
}
