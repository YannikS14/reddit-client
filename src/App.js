import React from 'react';
import './App.css';
import Header from './features/header/Header';
import PostList from './features/postList/PostList';
import SubredditList from './features/subredditList/SubredditList';

function App() {
  return (
    <div className="App bg-gray-100 min-h-screen">
      <Header />
      <main class="container mx-auto pt-20 flex flex-col sm:flex-row">
        <PostList
          posts={[
            {
              title: 'postTitle',
              votes: '35.8k',
              content: 'the content',
              postedBy: 'author',
              publishDate: '8 hours ago',
              commentsAmount: 345,
            },
            {
              title: 'postTitle',
              votes: '35.8k',
              content: 'the content',
              postedBy: 'author',
              publishDate: '8 hours ago',
              commentsAmount: 345,
            },
            {
              title: 'postTitle',
              votes: '35.8k',
              content: 'the content',
              postedBy: 'author',
              publishDate: '8 hours ago',
              commentsAmount: 345,
            },
          ]}
        />
        <SubredditList
          subreddits={[
            {
              icon: 'https://b.thumbs.redditmedia.com/EndDxMGB-FTZ2MGtjepQ06cQEkZw_YQAsOUudpb9nSQ.png',
              title: 'AskReddit',
            },
            {
              icon: 'https://b.thumbs.redditmedia.com/0PgZl68jAxA6T1BH6uvUQ5Bz1F1GrrJLCL8oi2Gz0Ak.png',
              title: 'gaming',
            },
          ]}
        />
      </main>
    </div>
  );
}

export default App;
