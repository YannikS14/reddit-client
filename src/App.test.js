import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import { store } from './app/store';
import Header from './features/header/Header';
import Posts from './features/posts/Posts';
import Subreddits from './features/subreddits/Subreddits';

describe('App component', () => {
  it('renders a <Header /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Header)).toHaveLength(1);
  });

  it('renders a <Posts /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Posts)).toHaveLength(1);
  });

  it('renders a <Subreddits /> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(Subreddits)).toHaveLength(1);
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
