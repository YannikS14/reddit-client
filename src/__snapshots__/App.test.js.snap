// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App bg-gray-100 dark:bg-gray-600 dark:text-gray-50 min-h-screen"
>
  <header
    className="fixed z-20 min-w-full shadow-lg bg-white dark:bg-gray-700 dark:text-gray-50"
  >
    <div
      className="container mx-auto px-4 h-20 flex justify-between items-center"
    >
      <div
        className="flex items-center"
        id="logo"
      >
        <img
          alt="reddit logo"
          className="w-8 fill-current text-green-600"
          src="reddit-logo.svg"
        />
        <h1
          className="pl-2 sm:text-xl font-semibold"
        >
          <span
            className="font-bold"
          >
            Reddit
          </span>
           Client
        </h1>
      </div>
      <div
        className="flex flex-1 mx-4 max-w-sm relative"
        id="search"
      >
        <input
          className="flex-1 w-full border-solid border-2 dark:bg-gray-600 border-gray-300 dark:border-gray-500 rounded-md pl-10 pr-4 py-2"
          onChange={[Function]}
          placeholder="Search Reddit"
          type="search"
          value=""
        />
        <svg
          className="h-5 w-5 absolute left-4 top-3 text-gray-700 dark:text-gray-300"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
          />
        </svg>
      </div>
      <div
        className="hidden sm:flex border-solid border-2 border-gray-300 dark:border-gray-500 rounded-md divide-x-2 divide-gray-300 dark:divide-gray-500 divide-solid"
        id="dark-mode-toggle"
      >
        <button
          className="px-4 py-2 text-primary"
          id="light-mode"
          onClick={[Function]}
        >
          <svg
            className="h-6 w-6 pointer-events-none"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
        <button
          className="px-4 py-2"
          id="dark-mode"
          onClick={[Function]}
        >
          <svg
            className="h-6 w-6 pointer-events-none"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
  <main
    className="container mx-auto pt-24 flex flex-col lg:flex-row"
  >
    <section
      className="lg:w-2/3 mx-4 flex-auto"
    >
      <article
        className="mb-8 p-6 bg-white text-gray-800 dark:bg-gray-700 dark:text-gray-50 rounded-md shadow-lg hover:shadow-xl transition-shadow duration-300 flex z-20 "
      >
        <div
          className="pr-6 flex flex-col items-center"
          id="votes"
        >
          <button
            aria-label="Upvote"
            className="text-gray-500 dark:text-gray-300 hover:text-green-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
          <p
            className="my-1.5 text-lg font-bold"
          >
            0
          </p>
          <button
            aria-label="Downvote"
            className="text-gray-500 dark:text-gray-300 hover:text-red-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 rotate-180 
            "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="flex-1 overflow-hidden"
          id="post-content"
        >
          <h2
            className="mb-4 text-xl font-semibold cursor-pointer"
            onClick={[Function]}
          >
            
          </h2>
          <span>
             
          </span>
          <hr
            className="mt-4"
          />
          <div
            className="flex flex-wrap justify-between px-8 py-2 gap-2"
            id="post-footer"
          >
            <div
              className="font-semibold text-sm"
            >
              
            </div>
            <div
              className="text-xs"
            >
              52 years ago
            </div>
            <div
              className="flex text-sm"
            >
              <button
                aria-label="Comments"
                className="text-gray-500 dark:text-gray-300 hover:text-gray-800 dark:hover:text-gray-50"
                onClick={[Function]}
              >
                <svg
                  className="h-6 w-6 pointer-events-none mr-1 transition duration-300"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 8h10M7 12h4m1 8l-4-4H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-3l-4 4z"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                  />
                </svg>
              </button>
              0
            </div>
          </div>
        </div>
      </article>
      <article
        className="mb-8 p-6 bg-white text-gray-800 dark:bg-gray-700 dark:text-gray-50 rounded-md shadow-lg hover:shadow-xl transition-shadow duration-300 flex z-20 "
      >
        <div
          className="pr-6 flex flex-col items-center"
          id="votes"
        >
          <button
            aria-label="Upvote"
            className="text-gray-500 dark:text-gray-300 hover:text-green-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
          <p
            className="my-1.5 text-lg font-bold"
          >
            0
          </p>
          <button
            aria-label="Downvote"
            className="text-gray-500 dark:text-gray-300 hover:text-red-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 rotate-180 
            "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="flex-1 overflow-hidden"
          id="post-content"
        >
          <h2
            className="mb-4 text-xl font-semibold cursor-pointer"
            onClick={[Function]}
          >
            
          </h2>
          <span>
             
          </span>
          <hr
            className="mt-4"
          />
          <div
            className="flex flex-wrap justify-between px-8 py-2 gap-2"
            id="post-footer"
          >
            <div
              className="font-semibold text-sm"
            >
              
            </div>
            <div
              className="text-xs"
            >
              52 years ago
            </div>
            <div
              className="flex text-sm"
            >
              <button
                aria-label="Comments"
                className="text-gray-500 dark:text-gray-300 hover:text-gray-800 dark:hover:text-gray-50"
                onClick={[Function]}
              >
                <svg
                  className="h-6 w-6 pointer-events-none mr-1 transition duration-300"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 8h10M7 12h4m1 8l-4-4H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-3l-4 4z"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                  />
                </svg>
              </button>
              0
            </div>
          </div>
        </div>
      </article>
      <article
        className="mb-8 p-6 bg-white text-gray-800 dark:bg-gray-700 dark:text-gray-50 rounded-md shadow-lg hover:shadow-xl transition-shadow duration-300 flex z-20 "
      >
        <div
          className="pr-6 flex flex-col items-center"
          id="votes"
        >
          <button
            aria-label="Upvote"
            className="text-gray-500 dark:text-gray-300 hover:text-green-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
          <p
            className="my-1.5 text-lg font-bold"
          >
            0
          </p>
          <button
            aria-label="Downvote"
            className="text-gray-500 dark:text-gray-300 hover:text-red-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 rotate-180 
            "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="flex-1 overflow-hidden"
          id="post-content"
        >
          <h2
            className="mb-4 text-xl font-semibold cursor-pointer"
            onClick={[Function]}
          >
            
          </h2>
          <span>
             
          </span>
          <hr
            className="mt-4"
          />
          <div
            className="flex flex-wrap justify-between px-8 py-2 gap-2"
            id="post-footer"
          >
            <div
              className="font-semibold text-sm"
            >
              
            </div>
            <div
              className="text-xs"
            >
              52 years ago
            </div>
            <div
              className="flex text-sm"
            >
              <button
                aria-label="Comments"
                className="text-gray-500 dark:text-gray-300 hover:text-gray-800 dark:hover:text-gray-50"
                onClick={[Function]}
              >
                <svg
                  className="h-6 w-6 pointer-events-none mr-1 transition duration-300"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 8h10M7 12h4m1 8l-4-4H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-3l-4 4z"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                  />
                </svg>
              </button>
              0
            </div>
          </div>
        </div>
      </article>
      <article
        className="mb-8 p-6 bg-white text-gray-800 dark:bg-gray-700 dark:text-gray-50 rounded-md shadow-lg hover:shadow-xl transition-shadow duration-300 flex z-20 "
      >
        <div
          className="pr-6 flex flex-col items-center"
          id="votes"
        >
          <button
            aria-label="Upvote"
            className="text-gray-500 dark:text-gray-300 hover:text-green-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
          <p
            className="my-1.5 text-lg font-bold"
          >
            0
          </p>
          <button
            aria-label="Downvote"
            className="text-gray-500 dark:text-gray-300 hover:text-red-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 rotate-180 
            "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="flex-1 overflow-hidden"
          id="post-content"
        >
          <h2
            className="mb-4 text-xl font-semibold cursor-pointer"
            onClick={[Function]}
          >
            
          </h2>
          <span>
             
          </span>
          <hr
            className="mt-4"
          />
          <div
            className="flex flex-wrap justify-between px-8 py-2 gap-2"
            id="post-footer"
          >
            <div
              className="font-semibold text-sm"
            >
              
            </div>
            <div
              className="text-xs"
            >
              52 years ago
            </div>
            <div
              className="flex text-sm"
            >
              <button
                aria-label="Comments"
                className="text-gray-500 dark:text-gray-300 hover:text-gray-800 dark:hover:text-gray-50"
                onClick={[Function]}
              >
                <svg
                  className="h-6 w-6 pointer-events-none mr-1 transition duration-300"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 8h10M7 12h4m1 8l-4-4H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-3l-4 4z"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                  />
                </svg>
              </button>
              0
            </div>
          </div>
        </div>
      </article>
      <article
        className="mb-8 p-6 bg-white text-gray-800 dark:bg-gray-700 dark:text-gray-50 rounded-md shadow-lg hover:shadow-xl transition-shadow duration-300 flex z-20 "
      >
        <div
          className="pr-6 flex flex-col items-center"
          id="votes"
        >
          <button
            aria-label="Upvote"
            className="text-gray-500 dark:text-gray-300 hover:text-green-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
          <p
            className="my-1.5 text-lg font-bold"
          >
            0
          </p>
          <button
            aria-label="Downvote"
            className="text-gray-500 dark:text-gray-300 hover:text-red-500"
            onClick={[Function]}
          >
            <svg
              className="h-6 w-6 pointer-events-none transform transition duration-300 rotate-180 
            "
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 11l3-3m0 0l3 3m-3-3v8m0-13a9 9 0 110 18 9 9 0 010-18z"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
          </button>
        </div>
        <div
          className="flex-1 overflow-hidden"
          id="post-content"
        >
          <h2
            className="mb-4 text-xl font-semibold cursor-pointer"
            onClick={[Function]}
          >
            
          </h2>
          <span>
             
          </span>
          <hr
            className="mt-4"
          />
          <div
            className="flex flex-wrap justify-between px-8 py-2 gap-2"
            id="post-footer"
          >
            <div
              className="font-semibold text-sm"
            >
              
            </div>
            <div
              className="text-xs"
            >
              52 years ago
            </div>
            <div
              className="flex text-sm"
            >
              <button
                aria-label="Comments"
                className="text-gray-500 dark:text-gray-300 hover:text-gray-800 dark:hover:text-gray-50"
                onClick={[Function]}
              >
                <svg
                  className="h-6 w-6 pointer-events-none mr-1 transition duration-300"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7 8h10M7 12h4m1 8l-4-4H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-3l-4 4z"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                  />
                </svg>
              </button>
              0
            </div>
          </div>
        </div>
      </article>
    </section>
    <section
      className="lg:mx-4 lg:w-1/3 lg:ml-0 flex-auto"
    >
      <div
        className="p-6 bg-white dark:bg-gray-700 sm:rounded-md shadow-lg"
      >
        <h2
          className="text-xl font-bold"
        >
          Subreddits
        </h2>
        <div
          className="mt-4"
        >
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary bg-gray-100 dark:bg-gray-600"
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://api.adorable.io/avatars/25/Home"
            />
            <p
              className="font-semibold"
            >
              Popular
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://b.thumbs.redditmedia.com/EndDxMGB-FTZ2MGtjepQ06cQEkZw_YQAsOUudpb9nSQ.png"
            />
            <p
              className="font-semibold"
            >
              AskReddit
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://a.thumbs.redditmedia.com/nmh5l-zCsmmc3y2ehfjtWRJjGmCEWEJDTjtW3AGMz60.png"
            />
            <p
              className="font-semibold"
            >
              Wallstreetbets
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://b.thumbs.redditmedia.com/0PgZl68jAxA6T1BH6uvUQ5Bz1F1GrrJLCL8oi2Gz0Ak.png"
            />
            <p
              className="font-semibold"
            >
              Gaming
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://b.thumbs.redditmedia.com/Kl3TBjINRBLd9sukJaSPts_0geISdO-jtVniyfCw1GA.png"
            />
            <p
              className="font-semibold"
            >
              CryptoCurrency
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://a.thumbs.redditmedia.com/4SKK4rzvSSDPLWbx4kt0BvE7B-j1UQBLZJsNCGgMz54.png"
            />
            <p
              className="font-semibold"
            >
              Germany
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://styles.redditmedia.com/t5_2twpw/styles/communityIcon_8bhyo9k303x71.png"
            />
            <p
              className="font-semibold"
            >
              Blackcats
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://api.adorable.io/avatars/25/Home"
            />
            <p
              className="font-semibold"
            >
              Cooking
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://b.thumbs.redditmedia.com/B7IpR8P1mEsQIjdizK5x79s5aGfJUtKk3u2ksGZ9n2Q.png"
            />
            <p
              className="font-semibold"
            >
              Todayilearned
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://styles.redditmedia.com/t5_2xiqj/styles/communityIcon_zvmcowk0d0671.jpg?format=pjpg&s=1537e0d3fd8639bce301c389ef60e99f44587d0a"
            />
            <p
              className="font-semibold"
            >
              Aoe4
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://styles.redditmedia.com/t5_2qgzy/styles/communityIcon_rvt3zjh1fc551.png"
            />
            <p
              className="font-semibold"
            >
              Sports
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://a.thumbs.redditmedia.com/E0Bkwgwe5TkVLflBA7WMe9fMSC7DV2UOeff-UpNJeb0.png"
            />
            <p
              className="font-semibold"
            >
              News
            </p>
          </button>
          <button
            className="w-full flex items-center p-4 rounded-md transition-colors hover:text-primary "
            onClick={[Function]}
          >
            <img
              alt=""
              className="w-6 h-6 mr-2 border border-gray-500 rounded-full"
              src="https://styles.redditmedia.com/t5_2s9h3/styles/communityIcon_ujdrcdijabb61.png"
            />
            <p
              className="font-semibold"
            >
              DunderMifflin
            </p>
          </button>
        </div>
      </div>
    </section>
  </main>
</div>
`;
